### PR TITLE
Created SmutPuppet network scraper

### PR DESCRIPTION
**Pull request by echo6ix**

### CODE REVIEW

File: scrapers/SmutPuppet.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  - action: scrapeXPath
+    url: 
+      - blackandbig.com/update
+      - darksodomy.com/update
+      - dothewife.com/update
+      - dreamtranny.com/update
+      - genlez.com/update
+      - goldenslut.com/update
+      - grannyvsbbc.com/update
+      - jefssmodels.com/update
```
Comment (Maista6969) [inline scrapers/SmutPuppet.yml:12]: Small typo here, should be `jeffsmodels.com/update`

File: scrapers/SmutPuppet.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+              - regex: .+\/updates/(\d+)\/.+
+                with: $1
+      # Return the sanitized URL
+      URL:
+        selector: //div[@class="block-logo"]/a/img/@src|//div[@class="model-player"]/a/img/@src
+        concat: "|"
+        postProcess:
+          - replace:
+              - regex: \/static\/(\w+\....)\/.+\/(\d+)\/.+
+                with: https://$1/update/$2
```
Comment (Maista6969) [inline scrapers/SmutPuppet.yml:65]: This works for every site except DoTheWife and Jeff'sModels

In addition the URL should include the trailing slash in order to be canonical; without this the site will redirect to the version with the slash anyway, so it's annoying when you rescrape a scene with a correct URL and get an additional nearly-identical URL in the list of URLs

The regex should be `https://$1/update/$2/`

**Review by Maista6969 — CHANGES_REQUESTED** [review]
Tested with
* https://blackandbig.com/update/1329/ - :heavy_check_mark:
* https://darksodomy.com/update/1162/ - :heavy_check_mark:
* https://dothewife.com/update/2274/ - Missing date and URL
* https://genlez.com/update/1674/ - :heavy_check_mark:
* https://goldenslut.com/update/1318/ - :heavy_check_mark:
* https://grannyvsbbc.com/update/1343/ - :heavy_check_mark:
* https://jeffsmodels.com/update/1125/ - Produces incomplete URL
* https://milfsodomy.com/update/1070/ - :heavy_check_mark:
* https://smutmerchants.com/update/1486/ - :heavy_check_mark:
* https://suggabunny.com/update/813/ - :heavy_check_mark:
* https://teenerotica.xxx/update/1623/ - :heavy_check_mark:

Of the sites you've added, `jeffsmodels.com` and `dothewife.com` already exist: in order to avoid conflicts you should delete the old scrapers (`DoTheWife.yml` and `JeffsModels.yml`) and update their entries in `SCRAPERS-LIST.md`

The remaining sites need their own entries in `SCRAPERS-LIST.md`

**Comment by echo6ix** [comment]
I've come across a glitch when scraping the studio code. If the studio code is less than 4 digits, the code will be scraped with a leading 0.

Example:

- https://dothewife.com/update/192/
  - xpath --> `//div[@class="model-player"]/a/img/@src`
  - returns --> `/usermedia/smutpuppet.com/tour03/updates/0192/01.jpg`
  - incorrectly resulting in a code of `0192` after postProcess
 
Adding a fix to my to-do list

**Comment by Maista6969** [comment]
A greedy zero in the regex should fix that: `\/static\/(\w+\....)\/.+\/0*(\d+)\/.+`

File: scrapers/SmutPuppet.yml (hunk 1 above)
Comment (echo6ix) [reply scrapers/SmutPuppet.yml]: Fixed

File: scrapers/SmutPuppet.yml (hunk 2 above)
Comment (echo6ix) [reply scrapers/SmutPuppet.yml]: Fixed both issues.

**Pull request merged**